src/dotnet/commands/dotnet-build3/Program.cs

```diff
@@ -1,19 +1,87 @@
 
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
 using Microsoft.DotNet.Cli.Utils;
-using System;
-using System.Diagnostics;
-using System.IO;
-using System.Reflection;
 
-namespace Microsoft.DotNet.Cli
+namespace Microsoft.DotNet.Tools.Build3
 {
     public class Build3Command
     {
         public static int Run(string[] args)
         {
-            return new MSBuildForwardingApp(args).Execute();
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            CommandLineApplication app = new CommandLineApplication(throwOnUnexpectedArg: false);
+            app.Name = "dotnet build3";
+            app.FullName = ".NET Builder";
+            app.Description = "Builder for the .NET Platform. Delegates to the MSBuild 'Build' target in the project file.";
+            app.AllowArgumentSeparator = true;
+            app.HelpOption("-h|--help");
+
+            CommandArgument projectArgument = app.Argument("<PROJECT>",
+                "The MSBuild project file to build. If a project file is not specified," +
+                " MSBuild searches the current working directory for a file that has a file extension that ends in `proj` and uses that file.");
+
+            CommandOption outputOption = app.Option("-o|--output <OUTPUT_DIR>", "Directory in which to place outputs", CommandOptionType.SingleValue);
+            CommandOption frameworkOption = app.Option("-f|--framework <FRAMEWORK>", "Compile a specific framework", CommandOptionType.SingleValue);
+            CommandOption configurationOption = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
+            CommandOption versionSuffixOption = app.Option("--version-suffix <VERSION_SUFFIX>", "Defines the value for the $(VersionSuffix) property in the project", CommandOptionType.SingleValue);
+
+            CommandOption noIncrementalOption = app.Option("--no-incremental", "Set this flag to turn off incremental build", CommandOptionType.NoValue);
+            CommandOption noDependenciesOption = app.Option("--no-dependencies", "Set this flag to ignore project to project references and only build the root project", CommandOptionType.NoValue);
+
+            app.OnExecute(() =>
+            {
+                List<string> msbuildArgs = new List<string>();
+
+                if (!string.IsNullOrEmpty(projectArgument.Value))
+                {
+                    msbuildArgs.Add(projectArgument.Value);
+                }
+
+                if (noIncrementalOption.HasValue())
+                {
+                    msbuildArgs.Add("/t:Rebuild");
+                }
+                else
+                {
+                    msbuildArgs.Add("/t:Build");
+                }
+
+                if (outputOption.HasValue())
+                {
+                    msbuildArgs.Add($"/p:OutputPath={outputOption.Value()}");
```

Comment:
Yes, it actually does.  This is because the whole string is added as a single arg string to `msbuildArgs`.  And when this enumerable of strings gets passed into `ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart` https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/ForwardingApp.cs#L75, ArgumentEscaper ensures the single string is treated as one argument to the process being started.